docker/env/build_n_push.sh

```diff
@@ -25,7 +25,7 @@ else
     echo "Hydra image with version $VERSION not found locally. Building..."
     cd "${DOCKER_ENV_DIR}"
     REQUIREMENTS_IN=$(realpath --relative-to=${DOCKER_ENV_DIR} ${SCT_DIR}/requirements.in)
-    pip-compile $REQUIREMENTS_IN --allow-unsafe --generate-hashes --resolver=backtracking > ${SCT_DIR}/${PY_PREREQS_FILE}
+    uv pip compile $REQUIREMENTS_IN --generate-hashes > ${SCT_DIR}/${PY_PREREQS_FILE}
```

Comment:
need to also update requirements.txt in this way to be in line. Also neest to update dev setup docs

Reply:
The `requirements.txt` gets updated exactly here, it is referenced by the `PY_PREREQS_FILE` env var.